core/src/main/java/hudson/model/DownloadService.java

```diff
@@ -302,6 +305,20 @@ public String getUrl() {
         }
 
         /**
+         * URLs to download from.
+         */
+        public List<String> getUrls() {
+            List<String> updateSites = new ArrayList<String>();
+            for (UpdateSite site : Jenkins.getActiveInstance().getUpdateCenter().getSiteList()) {
+                String siteUrl = site.getUrl();
+                int baseUrlEnd = siteUrl.indexOf("update-center.json");
```

Comment:
:ant: May return crappy results or even fail if the UC configuration does not contain "update-center.json" in the path. I would skip such URLs and print warnings